src/Agent.Worker/Release/ArtifactProvider.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Agent.Worker.Release.Artifacts.Definition;
+
+using Microsoft.VisualStudio.Agent.Worker.Release.Artifacts;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker.Release.Artifacts;
+using Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Contracts;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
+{
+    [ServiceLocator(Default = typeof(ArtifactProvider))]
+    public interface IArtifactProvider : IAgentService
+    {
+        Task Download(IExecutionContext executionContext, ArtifactDefinition artifactDefinition, string downloadFolderPath);
+
+        IArtifactDetails GetArtifactDetails(IExecutionContext context, AgentArtifactDefinition agentArtifactDefinition);
+    }
+
+    public class ArtifactProvider : AgentService, IArtifactProvider
+    {
+        public async Task Download(IExecutionContext executionContext, ArtifactDefinition artifactDefinition, string workingFolder)
+        {
+            Trace.Entering();
+
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            ArgUtil.NotNull(artifactDefinition, nameof(artifactDefinition));
+            ArgUtil.NotNullOrEmpty(workingFolder, nameof(workingFolder));
+
+            try
+            {
+                // TODO: Avoid this if-else case by implementing Custom ServiceLocator attribute
+                if (artifactDefinition.ArtifactType == ArtifactType.Build)
```

Comment:
I have removed the enum itself.. its  unncessary.. its a copy of AgentArtifactType.. I am directly using AgentArtifactType..